app/questionnaire/path_finder.py

```diff
@@ -178,3 +200,23 @@ def _remove_rule_answers(self, goto_rule, this_location):
                     self.answer_store.remove_answer(condition["id"])
 
         self.progress_store.remove_completed_location(location=this_location)
+
+
+def should_goto_new(
+    rule, schema, answer_store, list_store, metadata, location, routing_path_block_ids
+):
+    when_rule_evaluator = WhenRuleEvaluator(
+        schema,
+        answer_store,
+        list_store,
+        metadata,
+        location,
+        routing_path_block_ids,
+    )
+    if when_rule := rule.get("when", None):
+        operator = list(when_rule.keys())[0]
+        return when_rule_evaluator.evaluate(
+            rule={operator: [when_rule[operator][1], when_rule[operator][0]]}
+        )
```

Comment:
This shouldn't be necessary, it should work by passing the rule directly to the when rule evaluator i.e. `when_rule_evaluator.evaluate(when_rule)`. That doesn't work because when we deserialise schemas we convert lists to Tuples and there is a type check in the `evaluate` method that expects a list. That type check should allow a list or Tuple.

Reply:
This one is done now, `evaluate` allows tuple and only `when_rule` is passed, thanks!